### ISSUE TITLE
High memory usage when open html and ts file

### ISSUE THREAD

**Issue by mamontenegro1694** [angular/vscode-ng-language-service#712]
When i starts Vs Code it consumes like 300MB of Ram, but when i open a html file and ts file, angular language servies and js/ts services starts. Memory used when it happens is almost 800 MB only for those 2 services.

I don't know if it is because of my SO (Ubuntu 18.04) or something else.

Please help.

Sorry for my english, I'm from Colombia.

**Comment by ayazhafiz** [comment]
Could you create a CPU profile of when this happens? Instructions here: https://github.com/microsoft/vscode/wiki/Performance-Issues#profile-the-running-extensions

Or, can you provide us with an example project and what you did when you observed the high memory usage?

________________________________
От: Miguel Montenegro <notifications@github.com>
Отправлено: Monday, March 30, 2020 8:14:37 PM
Кому: angular/vscode-ng-language-service <vscode-ng-language-service@noreply.github.com>
Копия: Subscribed <subscribed@noreply.github.com>
Тема: [angular/vscode-ng-language-service] High memory usage when open html and ts file (#712)


When i starts Vs Code it consumes like 300MB of Ram, but when i open a html file and ts file, angular language servies and js/ts services starts. Memory used when it happens is almost 800 MB only for those 2 services.

I don't know if it is because of my SO (Ubuntu 18.04) or something else.

Please help.

Sorry for my english, I'm from Colombia.

**Comment by paxapy** [comment]
i've noticed the same problem, but in my case it was about 2Gb RAM only by ng-template extensions. yet it wasn't work :trollface: 
Ubuntu 18.04.4 LTS/code 1.44.1/Angular Language Service v0.901.0

**Comment by hoang-innomizetech** [comment]
I also encounter high memory usage by this extension (up to 2GB RAM).  See below process when using code --status command

![image](https://user-images.githubusercontent.com/43955692/80185318-5b84b080-8636-11ea-9553-528d0ed59b37.png)

We have a monorepo that contains a lot of files and folders but only a few folders (maybe only one) for Angular. I think adding an include/exclude settings will reduce the number of files to watch.

**Comment by emjimadhu** [comment]
I am having the same issue. But something weird happens with me. Whenever i open non angular project TS files, This plugin is ggetting initialized, i am sorry if my queestion is stupid. But is this the expected behaviour?

**Comment by ayazhafiz** [comment]
> But something weird happens with me. Whenever i open non angular project TS files, This plugin is ggetting initialized

Definitely not a stupid question! Are these TS files in an Angular project or have nothing to do with Angular at all? Can you show the output of the plugin?

**Comment by emjimadhu** [comment]
@ayazhafiz Sorry for the late reply. Here is the screenshot of output window

![Screenshot from 2020-04-30 21-58-42](https://user-images.githubusercontent.com/15604962/80735204-f6492780-8b2d-11ea-97ce-6f1d4c04ac6d.png)

And here is the log file

```
▶ cat /home/gnanamadhu/.config/Code/logs/20200430T215645/exthost1/Angular.ng-template/nglangsvc.log            
Info 0    [21:57:13.123] Format host information updated
Info 1    [21:57:13.123] reload projects.
Info 2    [21:57:13.123] Structure before ensureProjectForOpenFiles:
Info 3    [21:57:13.124] Structure after ensureProjectForOpenFiles:
Info 4    [21:57:13.124] Host file extension mappings updated
Info 5    [21:57:13.155] Search path: /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/src
Info 6    [21:57:13.157] For info: /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/src/app.controller.ts :: Config file name: /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json
Info 7    [21:57:13.158] Opened configuration file /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json
Info 8    [21:57:13.240] Config: /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json : {
 "rootNames": [
  "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/src/app.controller.spec.ts",
  "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/src/app.controller.ts",
  "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/src/app.module.ts",
  "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/src/app.service.ts",
  "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/src/main.ts",
  "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/test/app.e2e-spec.ts"
 ],
 "options": {
  "module": 1,
  "declaration": true,
  "removeComments": true,
  "emitDecoratorMetadata": true,
  "experimentalDecorators": true,
  "target": 4,
  "sourceMap": true,
  "outDir": "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/dist",
  "baseUrl": "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local",
  "incremental": true,
  "configFilePath": "/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json"
 }
}
Info 9    [21:57:15.847] Loading global plugin @angular/language-service
Info 10   [21:57:15.847] Enabling plugin @angular/language-service from candidate paths: /home/gnanamadhu/.vscode/extensions/angular.ng-template-0.901.4/server/node_modules/@angular/language-service/bundles/language-service.umd.js,/home/gnanamadhu/.vscode/extensions/angular.ng-template-0.901.4/node_modules/typescript/lib/tsserverlibrary.js/../../..
Info 11   [21:57:15.848] Loading @angular/language-service from /home/gnanamadhu/.vscode/extensions/angular.ng-template-0.901.4/server/node_modules/@angular/language-service/bundles/language-service.umd.js (resolved to /home/gnanamadhu/.vscode/extensions/angular.ng-template-0.901.4/server/node_modules/@angular/language-service/bundles/language-service.umd.js/node_modules)
Info 12   [21:57:16.318] Plugin validation succeded
Info 13   [21:57:16.474] Starting updateGraphWorker: Project: /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json
Info 14   [21:57:27.699] Finishing updateGraphWorker: Project: /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json Version: 1 structureChanged: true Elapsed: 11225ms
Info 15   [21:57:27.699] Project '/home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json' (Configured) 
Info 16   [21:57:27.699] 	Files (365)

Info 17   [21:57:27.699] -----------------------------------------------
Info 18   [21:57:27.706] Starting updateGraphWorker: Project: /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json
Info 19   [21:57:27.748] Finishing updateGraphWorker: Project: /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json Version: 2 structureChanged: false Elapsed: 42ms
Info 20   [21:57:27.749] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 21   [21:57:27.761] Disabling language service for /home/gnanamadhu/Projects/learning/nestjs/mvc-session-auth-local/tsconfig.json because it is not an Angular project ('@angular/core/core.d.ts' could not be found). If you believe you are seeing this message in error, please reinstall the packages in your package.json.```

**Comment by sanoodles** [comment]
Would need to find a project that i can share for proper steps to reproduce...

But I can say that on Windows 10 I have the same issue. As soon as I open an Angular component's .html file within an Angular project (via Open Folder), and the Angular Language Service has initialized, this extension alone uses *482 MB* of "Memory" as per the vscode Process Explorer.

Not sure if feasible, but it would serve me to be able to enable/disable some extension features. Possibly I don't need much what uses most of that memory. What I want most is just the ability to navigate to a child component from a parent component's .html.

**Comment by kyliau** [comment]
We have significantly improved incremental compilation performance in the new Ivy-native language service, released in [v11.1.0](https://github.com/angular/vscode-ng-language-service/releases/tag/v11.1.0).
Please try it out the new version. If you run into similar problem, please open a new issue. I'll close this for now.
